File: sandbox.client/front/ts/model/ClientRecordsToSave.ts
import {PhoneNumber} from "./PhoneNumber";
import {Address} from "./Address";
import {Charm} from "./Charm";
import {Gender} from "./Gender";

export class ClientRecordsToSave {
  public id: string;
  public surname: string;
  public name: string;
  public patronymic: string;
  public charm: Charm;
  public gender: Gender;
  public dateOfBirth: string;
  public addressF: Address;
  public addressR: Address;
  public phoneNumbers: PhoneNumber[];
  public totalBalance: number/*int*/;
  public minBalance: number/*int*/;
  public maxBalance: number/*int*/;

  constructor() {
    this.surname = "";
    this.name = "";
    this.charm = new Charm();
     
    // this.gender = new Gender();
    this.addressF = new Address();
    this.addressR = new Address();
    this.phoneNumbers = [new PhoneNumber()];
  }

  public assign(o: any): ClientRecordsToSave {
    this.id = o.id;
    this.name = o.name;
    this.surname = o.surname;
    this.patronymic = o.patronymic;
    this.charm = o.charm;
    this.gender = o.gender;
    this.dateOfBirth = o.dateOfBirth;

    this.totalBalance = o.totalBalance;
    this.minBalance = o.minBalance;
    this.maxBalance = o.maxBalance;

    if (o.addressF) {
      let addressF = new Address();
      addressF.street = o.addressF.street;
      addressF.house = o.addressF.house;
      addressF.flat = o.addressF.flat;
      this.addressF = addressF;
    }

    if (o.addressR) {
      let addressR = new Address();
      addressR.street = o.addressR.street;
      addressR.house = o.addressR.house;
      addressR.flat = o.addressR.flat;
      this.addressR = addressR;
    }

    if (o.phoneNumbers) this.phoneNumbers = o.phoneNumbers;

    return this;
  }

  public static copy(a: any): ClientRecordsToSave {
    let ret = new ClientRecordsToSave();
    ret.assign(a);
    return ret;
  }
}

File: sandbox.controller/src/kz/greetgo/sandbox/controller/model/ClientRecordsToSave.java
package kz.greetgo.sandbox.controller.model;

import java.util.ArrayList;
import java.util.List;

public class ClientRecordsToSave {
  public String id;
  public String surname;
  public String name;
  public String patronymic;
  public Charm charm;
  public Gender gender;
  public String dateOfBirth;
  public Address addressF;
  public Address addressR;
  public List<PhoneNumber> phoneNumbers = new ArrayList<>();
  public double totalBalance;/**/
  public double minBalance;/**/
  public double maxBalance;/**/

  public String toXml() {
    StringBuilder sb = new StringBuilder();
    sb.append("<client id=\"").append(id).append("\">\n");
    if (surname != null) sb.append("  <surname value=\"").append(surname).append("\"/>\n");
    if (name != null) sb.append("  <name value=\"").append(name).append("\"/>\n");
    if (patronymic != null) sb.append("  <patronymic value=\"").append(patronymic).append("\"/>\n");
    if (dateOfBirth != null) sb.append("  <birth value=\"").append(dateOfBirth).append("\"/>\n");
    sb.append("  <address>\n");
    if (addressF != null) {
      sb.append("    <fact street=\"").append(addressF.street)
        .append("\" house=\"").append(addressF.house)
        .append("\" flat=\"").append(addressF.flat).append("\"/>\n");
    }
    if (addressR != null) {
      sb.append("    <register street=\"").append(addressR.street)
        .append("\" house=\"").append(addressR.house)
        .append("\" flat=\"").append(addressR.flat).append("\"/>\n");
    }
    sb.append("  </address>\n");
    if (phoneNumbers != null) {
      for (PhoneNumber phoneNumber : phoneNumbers) {
        switch (phoneNumber.phoneType) {
          case HOME:
            sb.append("  <homePhone>").append(phoneNumber.number).append("</homePhone>\n");
            break;
          case WORK:
            sb.append("  <workPhone>").append(phoneNumber.number).append("</workPhone>\n");
            break;
          case MOBILE:
            sb.append("  <mobilePhone>").append(phoneNumber.number).append("</mobilePhone>\n");
        }
      }
    }
    sb.append("</client>\n");
    return sb.toString();
  }

  public static void main(String[] args) {
    ClientRecordsToSave clientRecords = new ClientRecordsToSave();
    clientRecords.id = "asd";
    clientRecords.surname = "asd";
    clientRecords.name = "asd";
    clientRecords.addressF = new Address();
    clientRecords.addressF.street = "street";
    clientRecords.addressF.house = "B";
    clientRecords.addressF.flat = "23";
    clientRecords.addressR = new Address();
    clientRecords.addressR.street = "street reg";
    clientRecords.addressR.house = "regH";
    clientRecords.addressR.flat = "23r";
    clientRecords.phoneNumbers = new ArrayList<>();
    PhoneNumber phoneNumber = new PhoneNumber();
    phoneNumber.phoneType = PhoneType.HOME;
    phoneNumber.number = "7074129845";
    PhoneNumber phoneNumber2 = new PhoneNumber();
    phoneNumber2.phoneType = PhoneType.MOBILE;
    phoneNumber2.number = "7074129845";
    PhoneNumber phoneNumber3 = new PhoneNumber();
    phoneNumber3.phoneType = PhoneType.WORK;
    phoneNumber3.number = "7273357848 вн. 2115";
    clientRecords.phoneNumbers.add(phoneNumber);
    clientRecords.phoneNumbers.add(phoneNumber2);
    clientRecords.phoneNumbers.add(phoneNumber3);
    System.out.println(clientRecords.toXml());
  }
}
